www/base/src/app/builders/builders.tpl.jade

```diff
@@ -37,7 +39,10 @@
             span(ng-show="tags_filter.length > 0")
                 span.label.clickable.label-danger(ng-click="tags_filter = []") x
           th(style="width:20%px;") Workers
-      tr(ng-repeat='builder in builders | filter: isBuilderFiltered | orderBy: "name"')
+      ul(uib-pagination, total-items="builders.length",
+        ng-model="currentPage", class="pagination-sm", boundary-link-numbers="true",
+        max-size=10, items-per-page="page_size")
+      tr(ng-repeat='builder in builders | orderBy: "name" | limitTo: page_size :(currentPage-1)*page_size | filter: {name: searchQuery} | filter: isBuilderFiltered')
```

Comment:
put the filters first, and you will see the paging work much better
```suggestion
      tr(ng-repeat='builder in builders  | filter: {name: searchQuery} | filter: isBuilderFiltered' | orderBy: "name" | limitTo: page_size :(currentPage-1)*page_size)
```

Reply:
The search doesn't work on pages>=2 when I put the filter(s) first

Reply:
Does this happen because you try to print table of results as they would appear on the second/third page, which is empty anyway when you just have one single builder match the tag?

Reply:
Oh got it @tardyp . Now it works perfectly. Thank you 😄

Reply:
Angularjs filter is a bit magical